Move wal queue to its own repository and minor bug fixes. (#1994)
grafana/alloy: CHANGELOG.md

- Fixed an issue in the `prometheus.exporter.postgres` component that would leak goroutines when the target was not reachable (@dehaansa)

> - Moved the wal queue implementation to a separate repository to share with other projects. Also fixed issue with reloading
>   configuration and prometheus duplication of metrics. (@mattdurham)

Reviewer: ```suggestion
- Fixed issue with reloading configuration and prometheus metrics duplication in `prometheus.write.queue`. (@mattdurham)
```

Reviewer: I don't think it's user-facing that we moved to use a library.
